mysql/toddl.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+// schemaToDDL performs schema conversion from the source DB schema to
+// Spanner. It uses the source schema in conv.SrcSchema, and writes
+// the Spanner schema to conv.SpSchema.
+func schemaToDDL(conv *internal.Conv) error {
+	for _, srcTable := range conv.SrcSchema {
+		spTableName, err := internal.GetSpannerTable(conv, srcTable.Name)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't map source table %s to Spanner: %s", srcTable.Name, err))
+			continue
+		}
+		var spColNames []string
+		spColDef := make(map[string]ddl.ColumnDef)
+		conv.Issues[srcTable.Name] = make(map[string][]internal.SchemaIssue)
+		// Iterate over columns using ColNames order.
+		for _, srcColName := range srcTable.ColNames {
+			srcCol := srcTable.ColDefs[srcColName]
+			colName, err := internal.GetSpannerCol(conv, srcTable.Name, srcCol.Name, false)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Couldn't map source column %s of table %s to Spanner: %s", srcTable.Name, srcCol.Name, err))
+				continue
+			}
+			spColNames = append(spColNames, colName)
+			ty, issues := toSpannerType(conv, srcCol.Type.Name, srcCol.Type.Mods)
+			if len(srcCol.Type.ArrayBounds) > 1 {
+				ty = ddl.String{Len: ddl.MaxLength{}}
+				issues = append(issues, internal.MultiDimensionalArray)
+			}
+			// TODO: add issues for all elements of srcCol.Ignored.
```

Comment:
Can we remove this? or who should be responsible for this TODO? try to add the name, e.g., `TODO(hengfeng)`.

Reply:
There are `Check,Identity,Default,Exclusion, ForeignKey` srcCol.ignored values. We are adding only `Default,ForeignKey` in Issues. These constants are declared in `Ignored` struct in `schema.go` file. 
Need to add it for reporting purpose according to `TODO` mentioned in `postgres/toddl.go` file. Hence, I also added it as TODO here as well.
I have now mentioned it as TODO(hengfeng)